File: backend/src/main/java/com/documed/backend/notifications/Notification.java
package com.documed.backend.notifications;

import com.documed.backend.additionalservices.AdditionalService;
import com.documed.backend.visits.model.Visit;
import lombok.Data;
import lombok.NonNull;

@Data
public class Notification {
  private int id;
  @NonNull private NotificationStatus status;
  private Visit visit;
  private AdditionalService additionalService;
  private NotificationType type;
}

File: frontend/src/shared/types/Notification.ts
import { AdditionalServiceLite } from './AdditionalService';
import { NotificationStatus } from './enums';
import { VisitLite } from './Visit';

export interface Notification extends NotificationLite {
  visit: VisitLite | null;
  additionalService: AdditionalServiceLite | null;
}

export interface NotificationLite {
  id: number;
  status: NotificationStatus;
}
